### ISSUE TITLE
Op type not registered 'NoOp' from toco

### ISSUE THREAD

**Issue by xsr-ai** [tensorflow/tensorflow#19267]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: linux ubuntu 17.10
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: r1.6
- **Python version**: python 3.6.3
- **Bazel version (if compiling from source)**: bazel-0.11.1-installer-linux-x86_64.sh
- **GCC/Compiler version (if compiling from source)**: gcc/g++ 6.4.0
- **CUDA/cuDNN version**: CUDA:9.1 cuDNN 7.1
- **GPU model and memory**: gtx 1060 6GB
- **Exact command to reproduce**:
bazel run --config=opt --copt=-msse4.1 --copt=-msse4.2   //tensorflow/contrib/lite/toco:toco -- --input_file=/home/andy/data/models/Best_model/MFN.pb   --input_format=TENSORFLOW_GRAPHDEF   --output_format=TFLITE   --output_file=/home/andy/data/models/Best_model/MFN.tflite   --inference_type=FLOAT   --inference_input_type=FLOAT   --input_arrays=input   --output_arrays=embeddings   --input_shapes=1,112,112,3

### Describe the problem
I had trained and frozen a MobileNet_v2 pretrain model which is under slim architecture to tensorflow graphdef protocol, when I use  toco convert to tflite(command as flow item 1),  I got a error as flow item 2. hold for someone kindly help me solve it, thanks.

1. toco cmd
`bazel run --config=opt --copt=-msse4.1 --copt=-msse4.2   //tensorflow/contrib/lite/toco:toco -- --input_file=/home/andy/data/models/Best_model/MFN.pb   --input_format=TENSORFLOW_GRAPHDEF   --output_format=TFLITE   --output_file=/home/andy/data/models/Best_model/MFN.tflite   --inference_type=FLOAT   --inference_input_type=FLOAT   --input_arrays=input   --output_arrays=embeddings   --input_shapes=1,112,112,3`

2. error logs
`2018-05-14 19:35:27.695827: F tensorflow/core/graph/graph.cc:283] Non-OK-status: status status: Not found: Op type not registered 'NoOp' in binary running on andy. Make sure the Op and Kernel are registered in the binary running in this process.`

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by andrewginns** [comment]
Any update on this? I have the same problem

**Comment by xsr-ai** [comment]
@andrewginns no, i can't solve it, wait for help.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by nishanthkorupolu** [comment]
Any updates on this issue? Facing the same problem.

**Comment by andrewginns** [comment]
@nishanthkorupolu try to update your tensorflow. The toco tools didn't work for me on tf1.6 but work on tf1.8+

**Comment by nishanthkorupolu** [comment]
@andrewginns I'm currently using tensorflow 1.8.0 and it's showing the same error.

**Comment by anitha-v** [comment]
Can anyone having this issue share your model with us, so that we can reproduce the issue ?

**Comment by suharshs** [comment]
NoOp's are a control flow op often introduces by initializers. You should ensure your graph is a frozen eval graph before converting it to TFLite. 

If you can still produce this error with the most recent version of TF could you provide repro example?

**Comment by lkfo415579** [comment]
@suharshs  thanks for your reply. Here is my [repro example](https://github.com/lkfo415579/nmt_custom)(**Machine Translation Project origin from google group**, [origin_repro](https://github.com/tensorflow/nmt)), i modified some of the code in order make it simpler for lite converter project. 
Converting into frozen and lite model's code : [code](https://github.com/lkfo415579/nmt_custom/blob/041dd8ffe4afcaf0dfe96d86d8de17a079372fcc/nmt/inference.py#L213)

I have a few questions.

1. what are the different between python version toco(**lite.py etc. tf.contrib.lite.TocoConverter.from_frozen_graph**) and command version? which one is the newest ones?
2. I have already used **tf.graph_util.convert_variables_to_constants** python script to convert normal saved model into frozen graph but it still appears NoOp.
3. I tried **tf.graph_util.remove_training_nodes function** for removing training nodes(may contain NoOp?) but the model will be broken(unable to inference, it may change some of the Op of the old graph then it pops errors again.). strange thing is **NoOp** problem is passed during toco converting.

**Comment by cy89** [comment]
@aselle @andrehentz @petewarden would you PTAL?

**Comment by aselle** [comment]
@xsr-ai, could you verify that @suharshs' suggestions helped.

@lkfo415579 could you please open a new issue rather than posting on an existing one (it's difficult to track multiple conversations).

**Comment by tofulawrence** [comment]
No activity for @xsr-ai's concern. New issue opened for @lkfo415579.

**Comment by liukimnghia** [comment]
any one solve this problem?

**Comment by DoubangoTelecom** [comment]
Check https://github.com/DoubangoTelecom/TensorflowStaticNativeLibsAndroid/blob/master/README.md#known-issue
